packages/babel-parser/test/fixtures/estree/typescript/optional-chaining/output.json

```diff
@@ -51,6 +51,51 @@
             "optional": false
           }
         }
+      },
+      {
+        "type": "ExpressionStatement",
+        "start":15,"end":31,"loc":{"start":{"line":2,"column":0},"end":{"line":2,"column":16}},
+        "expression": {
+          "type": "ChainExpression",
+          "start":15,"end":30,"loc":{"start":{"line":2,"column":0},"end":{"line":2,"column":15}},
+          "expression": {
+            "type": "CallExpression",
```

Comment:
This should be an `OptionalCallExpression`.

Reply:
There is `optional: true` at the end of this depth. Is it different? 🤔

Reply:
Yes! `optional: true` means "here you have a question mark", while `OptionalCallExpression` means "there is a question mark somewhere".

For example, in `a?.()()` both are `OptionalCallExpression` but only the first one has `optional: true`.

Reply:
The behaviour is correct. The test enables the `estree` plugin, which converts `OptionalCallExpression` to a `ChainExpression`.

Reply:
Oh I didn't realize it.

Reply:
Thanks a lot! I patched the code, and other part is now `OptionalCallExpression`.